src/Workspaces/Remote/ServiceHub.CoreComponents/CoreComponents.Shared.targets

```diff
@@ -38,7 +38,21 @@
     <PackageReference Include="Microsoft.NETCore.App.crossgen2.win-x64" Condition="'$(GenerateReadyToRun)' == 'true'" ExcludeAssets="all" PrivateAssets="all" GeneratePathProperty="true" />    
   </ItemGroup>
   
-  <Target Name="CalculateCrossgenInputs" DependsOnTargets="PublishItemsOutputGroup">
+  <Target Name="CalculateCrossgenInputs" DependsOnTargets="PublishItemsOutputGroup">  
+    <PropertyGroup>
+      <!-- 
+        For BCL, we want to use the version provided by the runtime in VS, not the ones from the NuGet packages. 
+        This might not be possible if Roslyn is referencing a higher version than the one shipped as part of runtime in VS.
+        For example, we could be referecing S.C.I 7.0 but VS still ships .NET 6. Usually this is a transient state, and the
+        two would eventually be in sync. But we'd need to include those binaries during the out-of-sync period.
+        Note, for the same reasone, we can't safely exclude shared dependencies from ServiceHub host folder.
```

Comment:
Wouldn't this be a functional break if we got into this state? If roslyn is reference SCI at version 9.0.0.0 and VS runtime is still using 8.0.0.0 then we open ourselves to all kinds of type / member mismatch issues correct?

Reply:
if we ship SCI in our own vsix, then at runtime we'd load our SCI in our own ACL in the servicehub host process and thing would work.